### PR TITLE
Polish diagnostic messages

### PR DESCRIPTION
**Pull request by henrymercer**
This PR makes some improvements to the diagnostic messages for C#, Java, JavaScript, and Ruby to improve clarity, consistency, and fix some diagnostics that weren't parsing.

Review is particularly encouraged on the parts of the autobuilders that I have modified — feel free to pull out particular issues into separate PRs if it's easier to fix it yourself than suggest changes.  There will almost certainly also be things that I've missed — please make further updates either directly to this PR or separately.

Checks on this PR will fail until the corresponding internal PR has been merged.

### CODE REVIEW

File: ruby/extractor/src/diagnostics.rs [hunk 1]
```diff
@@ -92,7 +93,7 @@ impl LogWriter {
         DiagnosticMessage {
             timestamp: chrono::Utc::now(),
             source: Source {
-                id: format!("{}/{}", self.extractor, id),
+                id: format!("rb/{}", id),
```
Comment (aibaars) [inline ruby/extractor/src/diagnostics.rs:96]: The Ruby extractor is actually language agnostic, so having `rb` in this file is not great. I'll refactor this later.

**Review by aibaars — APPROVED** [review]
👍 for Ruby

**Review by erik-krogh — DISMISSED** [review]
JS 👍

**Review by smowton — DISMISSED** [review]
Java 👍

**Review by mbg — APPROVED** [review]

**Review by aibaars — CHANGES_REQUESTED** [review]
C# has test failures .

**Review by aibaars — DISMISSED** [review]

File: ruby/extractor/src/extractor.rs [hunk 2]
```diff
@@ -255,7 +255,7 @@ impl<'a> Visitor<'a> {
         );
         let mut mesg = self
             .diagnostics_writer
```
Comment (aibaars) [inline ruby/extractor/src/extractor.rs:257]:
```suggestion
        let mut mesg = self.diagnostics_writer.new_entry(
```

File: ruby/extractor/src/extractor.rs [hunk 3]
```diff
@@ -255,7 +255,7 @@ impl<'a> Visitor<'a> {
...
-            .new_entry("parse-error", "Parse error");
+            .new_entry("parse-error", "Could not process some files due to syntax errors");
```
Comment (aibaars) [inline ruby/extractor/src/extractor.rs:258]:
```suggestion
            "parse-error",
            "Could not process some files due to syntax errors",
        );
```

File: ruby/extractor/src/extractor.rs [hunk 4]
```diff
@@ -367,7 +367,7 @@ impl<'a> Visitor<'a> {
                 self.record_parse_error(
                     loc,
                     self.diagnostics_writer
-                        .new_entry("parse-error", "Parse error")
+                        .new_entry("parse-error", "Could not process some files due to syntax errors")
```
Comment (aibaars) [inline ruby/extractor/src/extractor.rs:370]:
```suggestion
                        .new_entry(
                            "parse-error",
                            "Could not process some files due to syntax errors",
                        )
```

**Review by aibaars — COMMENTED** [review]
Some formatting issues need fixing.

**Review by aibaars — APPROVED** [review]

**Pull request merged**